routing/router.go

```diff
@@ -1252,6 +1228,107 @@ func (r *ChannelRouter) networkHandler() {
 	}
 }
 
+func (r *ChannelRouter) getMissingBlocks(currentHeight uint32,
+	chainUpdate *chainview.FilteredBlock) {
+
+	outdatedHash, err := r.cfg.Chain.GetBlockHash(int64(currentHeight))
+	if err != nil {
+		log.Error("unable to retrieve outdated block hash")
+	}
+
+	outdatedBlock := &chainntnfs.BlockEpoch{
+		Height: int32(currentHeight),
+		Hash:   outdatedHash,
+	}
+
+	epochClient, err := r.cfg.Notifier.RegisterBlockEpochNtfn(
+		outdatedBlock,
+	)
+	if err != nil {
+		log.Error("unable to register notifications")
+	}
+	defer epochClient.Cancel()
+
+	blockDifference := int(chainUpdate.Height - currentHeight)
+
+	// We'll walk through all the outdated blocks and make sure we're able
+	// to update the graph with any closed channels from them.
+	for i := 0; i < blockDifference; i++ {
+		var missingBlock *chainntnfs.BlockEpoch
+
+		select {
+		case missingBlock = <-epochClient.Epochs:
+		case <-r.quit:
+			return
+		}
+
+		filteredBlock, err := r.cfg.ChainView.FilterBlock(
```

Comment:
If `filteredBlock` is `nil`, `updateGraphWithClosedChannels` will nil-pointer-deref

Reply:
Hmmm I see, since FilterBlock will return nil when shutting down? so should we add a nil check before moving forward here?

Reply:
nil-checking the `err` would fix it and at other call-sites.  it's only a preventative measure for the worst-case